### ISSUE TITLE
possible issue with large no of files

### ISSUE THREAD

**Issue by ajinabraham** [ajinabraham/njsscan#1]
```
les/xml/lib/xml.js', '../../Downloads/xvna-master/xvna/public/bower_components/Chart.js/src/Chart.Core.js', '../../Downloads/xvna-master/xvna/public/bower_components/morris.js/examples/days.html', '../../Downloads/xvna-master/xvna/public/bower_components/jquery/src/deprecated.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/node_modules/grunt/node_modules/js-yaml/node_modules/argparse/node_modules/underscore/LICENSE', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/node_modules/grunt/node_modules/lodash/vendor/tar/vendor/block-stream/LICENCE', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/istanbul/node_modules/escodegen/node_modules/source-map/.npmignore', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/LiveScript/lib/livescript.js', '../../Downloads/xvna-master/xvna/node_modules/lodash/sortBy.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/socket.io/node_modules/socket.io-client/node_modules/active-x-obfuscator/node_modules/zeparser/ZeParser.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/optimist/node_modules/wordwrap/test/break.js', '../../Downloads/xvna-master/xvna/node_modules/mime/mime.js', '../../Downloads/xvna-master/xvna/node_modules/lodash/_arrayLikeKeys.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-ui-mask/dist/mask.js', '../../Downloads/xvna-master/xvna/node_modules/lodash/fp/random.js', '../../Downloads/xvna-master/xvna/node_modules/exec/tests/timeout.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/socket.io/node_modules/socket.io-client/test/socket.test.js', '../../Downloads/xvna-master/xvna/public/bower_components/fullcalendar/dist/lang/id.js', '../../Downloads/xvna-master/xvna/public/bower_components/jquery-ui/ui/minified/i18n/jquery.ui.datepicker-fr.min.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-resource/index.js', '../../Downloads/xvna-master/xvna/public/bower_components/moment/src/lib/units/day-of-week.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/http-proxy/examples/node_modules/connect/lib/middleware/session/memory.js', '../../Downloads/xvna-master/xvna/public/bower_components/jquery-ui/ui/minified/i18n/jquery.ui.datepicker-sr-SR.min.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/node_modules/grunt/node_modules/js-yaml/node_modules/argparse/test/suppress.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/istanbul/test/browser/support/server.js', '../../Downloads/xvna-master/xvna/node_modules/lodash.get/index.js', '../../Downloads/xvna-master/xvna/public/bower_components/jquery-ui/ui/jquery.ui.resizable.js', '../../Downloads/xvna-master/xvna/node_modules/lodash/fp/forIn.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/glob/package.json', '../../Downloads/xvna-master/xvna/public/app/partials/app.html', '../../Downloads/xvna-master/xvna/public/bower_components/jquery_appear/bower.json', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/node_modules/grunt/node_modules/nopt/.npmignore', '../../Downloads/xvna-master/xvna/node_modules/lodash/fp/pathOr.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/istanbul/node_modules/fileset/LICENSE-MIT', '../../Downloads/xvna-master/xvna/public/bower_components/moment/locale/fy.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/log4js/node_modules/readable-stream/test/simple/test-stream2-read-sync-stack.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/socket.io/node_modules/socket.io-client/node_modules/uglify-js/test/unit/compress/expected/mangle.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/istanbul/node_modules/handlebars/node_modules/uglify-js/test/unit/compress/test/issue278.js', '../../Downloads/xvna-master/xvna/node_modules/lodash/fp/forOwnRight.js', '../../Downloads/xvna-master/xvna/public/bower_components/moment/src/lib/create/from-string-and-format.js', '../../Downloads/xvna-master/xvna/public/bower_components/flot/jquery.flot.resize.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/socket.io/node_modules/socket.io-client/node_modules/uglify-js/test/unit/compress/expected/issue9.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/node_modules/grunt/node_modules/js-yaml/node_modules/argparse/examples/arguments.js', '../../Downloads/xvna-master/xvna/public/bower_components/morris.js/examples/donut.html', '../../Downloads/xvna-master/xvna/public/bower_components/moment/src/locale/en-gb.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/istanbul/node_modules/escodegen/node_modules/source-map/Makefile.dryice.js', '../../Downloads/xvna-master/xvna/node_modules/lodash/_hashSet.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/http-proxy/examples/node_modules/connect-restreamer/.npmignore', '../../Downloads/xvna-master/xvna/node_modules/lodash/fp/some.js', '../../Downloads/xvna-master/xvna/public/bower_components/mocha/mocha.js', '../../Downloads/xvna-master/xvna/public/bower_components/plugins/pagination/four_button.js', '../../Downloads/xvna-master/xvna/public/bower_components/raphael/dev/amdDev.js', '../../Downloads/xvna-master/xvna/public/bower_components/jquery-ui/bower.json', '../../Downloads/xvna-master/xvna/public/bower_components/footable/demos/js/data-generator.js', '../../Downloads/xvna-master/xvna/node_modules/bson/lib/bson/parser/deserializer.js', '../../Downloads/xvna-master/xvna/public/bower_components/moment/src/locale/mr.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-ui-uploader/.gitignore', '../../Downloads/xvna-master/xvna/public/bower_components/moment/src/locale/hr.js', '../../Downloads/xvna-master/xvna/node_modules/lodash/flowRight.js', '../../Downloads/xvna-master/xvna/public/bower_components/rickshaw/vendor/d3.v3.js', '../../Downloads/xvna-master/xvna/public/bower_components/angular-count-to/scripts/node_modules/karma/node_modules/istanbul/test/other/test-collector.js', '../../Downloads/xvna-master/xvna/node_modules/lodash/fp/__.js', '../../Downloads/xvna-master/xvna/public/bower_components/chosen/chosen.jquery.js', '../../Downloads/xvna-master/xvna/public/bower_components/es5-shim/es5-sham.min.js']
(Pdb) c
Traceback (most recent call last):
  File "/Users/XXXXXX/njsscan/venv/bin/njsscan", line 11, in <module>
    load_entry_point('njsscan', 'console_scripts', 'njsscan')()
  File "/Users/XXXXXX/njsscan/njsscan/__main__.py", line 103, in main
    scan_results = NJSScan(args).scan()
  File "/Users/XXXXXX/njsscan/njsscan/njsscan.py", line 37, in scan
    result = scanner.scan()
  File "/Users/XXXXXX/njsscan/venv/src/libsast/libsast/scanner.py", line 63, in scan
    self.options).scan(valid_paths)
  File "/Users/XXXXXX/njsscan/venv/src/libsast/libsast/core_sgrep/semantic_sgrep.py", line 62, in scan
    self.format_output(sgrep_main.main(args))
  File "/Users/XXXXXX/njsscan/venv/src/semgrep/sgrep_lint/semgrep/sgrep_main.py", line 554, in main
    targeting_options=list(yield_targeting_options(args)),
  File "/Users/XXXXXX/njsscan/venv/src/semgrep/sgrep_lint/semgrep/sgrep_main.py", line 144, in invoke_sgrep
    output = subprocess.check_output(cmd, shell=False)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 411, in check_output
    **kwargs).stdout
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 488, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_fil
```

**Comment by prabhu** [comment]
Testing this repo with https://github.com/bkimminich/juice-shop

With nodejsscan, I had to do weird hacks like [this](https://github.com/ShiftLeftSecurity/sast-scan/blob/master/scan#L400) to make it run properly without hanging.

With this repo, I think the ignore_directories are not working quite well yet.

**Comment by prabhu** [comment]
It can successfully scan views, models, and routes directory in juice-shop. It hangs on the frontend directory which also contains node_modules.

The reporting json format is quite different to nodejsscan and is kind of difficult to work with. Since the project is quite new can you make the json match the nodejsscan format for now?

**Comment by ajinabraham** [comment]
How did you add ignore_directories. The supported settings is called `ignore-paths`
```yaml
ignore-paths:
  - __MACOSX
  - skip_dir
  - node_modules
```
See: https://github.com/ajinabraham/njsscan#configure-njsscan

I saw it hanging with the following exception

```
-c intercepted, will do some cleaning before exiting
[Parmap]: error at index j=3 in (10,14), chunksize=5 of a total of 5 got exception Common.UnixExit(-1) on core 2
an internal error occured while invoking the semgrep engine: unknown exception: Common.UnixExit(-1).
An error occurred while invoking the semgrep engine; please help us fix this by filing an an issue at https://semgrep.dev
```
This is probably a bug in semgrep. Discussing with the semgrep team. Also `node_modules` skipped by default.

I was able to scan it with the following file `juice-shop-master/.njsscan`

```yaml
---
- nodejs-extensions:
  - .js
  ignore-filenames:
  - dat.gui.min.js
  - three.js
```

This is the JSON reporting format we will be using from now on, All findings sorted by rule id. The previous report was bit naive and was a quick hack.

**Comment by prabhu** [comment]
I didn't add any configuration. Can you make those the default?

**Comment by ajinabraham** [comment]
Make what default?

**Comment by ajinabraham** [comment]
Fixed in 0.0.7
